common/str.cpp

```diff
@@ -55,6 +55,11 @@ String &String::operator=(const String &str) {
 	return *this;
 }
 
+String &String::operator=(String &&str) {
+	assign(static_cast<String &&>(str));
```

Comment:
Is the static cast needed here? That seems strange to cast a `String &&` into a `String &&`.

Reply:
Yes, that's how it works.  Even though the type is `String &&`, it's not treated as an rvalue when used in an expression, so it has to be cast.  Without the cast, it will pass to the `const String &` overload instead.